### PR TITLE
Conduit [Danylo Kunchenko]

### PR DESCRIPTION
**Pull request by d-kunchenko**

### CODE REVIEW

**Review by Iziumova — APPROVED** [review]

File: cypress.config.js [hunk 1]
```diff
@@ -12,10 +12,9 @@ module.exports = defineConfig({
     setupNodeEvents(on, config) {
       on("task", {
         generateUser() {
-          randomNumber = Math.ceil(Math.random(1000) * 1000);
           return {
-            username: faker.name.firstName() + `${randomNumber}`,
-            email: 'test'+`${randomNumber}`+'@mail.com',
+            username: faker.internet.userName(),
+            email: 'testuser@example.com',
```
Comment (YevhS) [inline cypress.config.js:17]: Email should be generated using faker instead of hardcoded value.

File: cypress/e2e/user.cy.js [hunk 2]
```diff
@@ -2,11 +2,23 @@
...
 
-  it.skip('should be able to follow the another user', () => {
-
-  });
+   it('should be able to follow the another user', () => {
+      cy.login(firstUser);
+      cy.wait(2000);
+      cy.visit(`/#/@${secondUser.username}`);
+      cy.getByDataQa('follow-btn').click();
+   });
```
Comment (YevhS) [inline cypress/e2e/user.cy.js:23]: Add assertions to check if the user is actually followed, e.g. by checking if the button text changes to 'Unfollow' or by checking the user's followers list.

File: cypress/e2e/user.cy.js [hunk 3]
```diff
@@ -2,11 +2,23 @@
...
+         secondUser = user;
+      });
+   });
 
-  it.skip('should be able to follow the another user', () => {
-
-  });
+   it('should be able to follow the another user', () => {
+      cy.login(firstUser);
+      cy.wait(2000);
```
Comment (YevhS) [inline cypress/e2e/user.cy.js:20]: Avoid using hardcoded wait times. Instead, use Cypress commands like 'cy.intercept()' to wait for specific API calls to complete.

File: cypress/e2e/settings.cy.js [hunk 4]
```diff
@@ -1,32 +1,76 @@
...
+   beforeEach(() => {
+      cy.task('db:clear');
+      cy.task('generateUser').then(generateUser => {
+         user = generateUser;
+      });
+      cy.login(user)
+      cy.visit('/#/settings');
+   });
 
-  });
```
Comment (YevhS) [inline cypress/e2e/settings.cy.js:11]: The first `beforeEach` block is empty and should be removed.

File: cypress/e2e/article.cy.js [hunk 5]
```diff
@@ -1,24 +1,163 @@
...
+   const errorMsg = 'Oops!';
+   const { title, description, body, tags } = generateNewArticle();
 
-  });
+   beforeEach(() => {
+      cy.task('db:clear');
+      cy.task('generateUser').then(generateUser => {
+         user = generateUser;
+      });
+      cy.register().then(user => {
```
Comment (YevhS) [inline cypress/e2e/article.cy.js:19]: Replace 'cy.login(user).then(() => user);' with 'cy.login(user);'. The 'then' statement is not necessary here.

File: cypress/e2e/article.cy.js [hunk 6]
```diff
@@ -1,24 +1,163 @@
...
+   });
 
-  });
+   it('should create an article with filled all fields', () => {
+      cy.findByPlaceholder('Article Title').type(title);
+      cy.findByPlaceholder('What\'s this article about?').type(description);
+      cy.findByPlaceholder('Write your article (in markdown)').type(body);
+      cy.findByPlaceholder('Enter tags').type(tags);
+      cy.getByDataQa('publish-btn').click();
+      cy.url(`articles/${title}-`);
```
Comment (YevhS) [inline cypress/e2e/article.cy.js:39]: Replace 'cy.url(`articles/${title}-`);' with 'cy.url().should('contain', `articles/${title}-`);' to correctly check the URL.

File: cypress/e2e/article.cy.js [hunk 7]
```diff
@@ -1,24 +1,163 @@
...
+      cy.get('h1').should('contain.text', title);
+   });
 
-  });
+   it('should create an article with empty Article Description field', () => {
+      cy.findByPlaceholder('Article Title').type(title);
+      cy.findByPlaceholder('Write your article (in markdown)').type(body);
+      cy.findByPlaceholder('Enter tags').type(tags);
+      cy.getByDataQa('publish-btn').click();
+      cy.url(`articles/${title}-`);
```
Comment (YevhS) [inline cypress/e2e/article.cy.js:57]: Replace 'cy.url(`articles/${title}-`);' with 'cy.url().should('contain', `articles/${title}-`);' to correctly check the URL.

File: cypress/e2e/settings.cy.js [hunk 8]
```diff
@@ -1,32 +1,76 @@
 /// <reference types="cypress" />
 /// <reference types="../support" />
 
+import { generateData } from "../support/generateData";
+
```
Comment (YevhS) [inline cypress/e2e/settings.cy.js:5]: just one empty line to separate

**Review by YevhS — APPROVED** [review]
you shouldn't push package-lock and package files

**Pull request closed without merging**